runtime/plugin/osc52.lua

```diff
@@ -17,28 +17,13 @@ require('vim.termcap').query('Ms', function(cap, found, seq)
 
   assert(cap == 'Ms')
 
-  -- Check 'clipboard' and g:clipboard again to avoid a race condition
-  if vim.o.clipboard ~= '' or vim.g.clipboard ~= nil then
-    return
-  end
-
   -- If the terminal reports a sequence other than OSC 52 for the Ms capability
   -- then ignore it. We only support OSC 52 (for now)
   if not seq or not seq:match('^\027%]52') then
     return
   end
 
-  local osc52 = require('vim.ui.clipboard.osc52')
-
-  vim.g.clipboard = {
-    name = 'OSC 52',
-    copy = {
-      ['+'] = osc52.copy('+'),
-      ['*'] = osc52.copy('*'),
-    },
-    paste = {
-      ['+'] = osc52.paste('+'),
-      ['*'] = osc52.paste('*'),
-    },
-  }
+  local termfeatures = vim.split(vim.g.termfeatures or '', ',', { trimempty = true })
+  termfeatures[#termfeatures + 1] = 'osc52'
+  vim.g.termfeatures = table.concat(termfeatures, ',')
```

Comment:
Using `g:termfeatures` here is a bit of a kludge so that we can communicate OSC 52 support to the clipboard.vim script.

I think a more elegant solution would be to add a new UI event for copying and pasting text. The TUI would implement this event using OSC 52, but other UIs can implement it using their own GUI toolkits. How do Neovim GUIs handle copy-paste today? Do they still shell out to external tools like the TUI does? If so then having a UI event seems like it would be a nice addition since GUIs should trivially be able to interface with the system clipboard.

cc @justinmk